realm/kotlin-extensions/src/androidTest/kotlin/io/realm/CoroutineTests.kt

```diff
@@ -207,10 +207,11 @@ class CoroutineTests {
                         assertTrue(flowResults.isFrozen)
                         assertEquals(0, flowResults.size)
                     }.onCompletion {
-                        countDownLatch.countDown()
-
-                        if (countDownLatch.count == 0L && !realmInstance.isClosed) {
+                        if (countDownLatch.count > 1) {
```

Comment:
Calling `countDown()` as the first thing could lead to finishing the test before properly closing the Realm.